### PR TITLE
MMSP-11484 (Mark/Eric) Add basic WinINet support

### PR DESCRIPTION
**Pull request by dgolub**
@markbenvenuto This adds basic support for downloading files using WinINet from Go.

@edaniels Feel free to weigh in or not as you see fit here.

#### QA Performed
* Used test driver program to download http://www.google.com using `OpenURL`
* Used test driver program to download http://www.google.com using `HTTPRequest`

### CODE REVIEW

File: internet.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+func (self *InternetSession) OpenURL(url string, headers string, flags URLRequestFlags) (*InternetFile, error) {
+	var headersRaw *uint16
+	if headers != "" {
+		headersRaw = syscall.StringToUTF16Ptr(headers)
+	}
+	handle, err := wrappers.InternetOpenUrl(
+		self.handle,
+		syscall.StringToUTF16Ptr(url),
+		headersRaw,
+		uint32(len(headers)),
```
Comment (markbenvenuto) [inline internet.go]: This should be the length of headersRaw

File: internet.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+	if version != "" {
+		versionRaw = syscall.StringToUTF16Ptr(version)
+	}
+	var refererRaw *uint16
+	if referer != "" {
+		refererRaw = syscall.StringToUTF16Ptr(referer)
+	}
+	var acceptTypesRaw **uint16
+	if acceptTypes != nil {
+		acceptTypesPtrs := make([]*uint16, len(acceptTypes))
```
Comment (markbenvenuto) [inline internet.go]: This array needs one additional member, and it needs to be NULL.

**Comment by markbenvenuto** [comment]
Done reviewing

File: internet.go (hunk 1 above)
Comment (dgolub) [inline internet.go]: No, this is right.  We want the length of the string in characters, not the size of the pointer to it.

File: internet.go (hunk 2 above)
Comment (dgolub) [inline internet.go]: Good catch.

**Comment by dgolub** [comment]
Merged 7b95bd0.

**Pull request closed without merging**